### PR TITLE
February rke2 r2

### PR DESCRIPTION
**Pull request by briandowns**
https://github.com/rancher/rancher/issues/36337

### CODE REVIEW

File: channels-rke2.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 channels:
   - name: default
-    latest: v1.21.10+rke2r1
+    latest: v1.21.10-rc2+rke2r1
```
Comment (rancher-max) [inline channels-rke2.yaml]: change this to be the `r2` at the end also

File: channels-rke2.yaml [hunk 2]
```diff
@@ -325,6 +325,33 @@ releases:
       rke2-canal:
         repo: rancher-rke2-charts
         version: v3.21.4-build2022020801
+  - version: v1.21.10-rc2+rke2r2
+    minChannelServerVersion: v2.6.1-alpha1
```
Comment (rancher-max) [inline channels-rke2.yaml]: I think this needs to be updated to be `v2.6.4-alpha1` per our last internal KDM discussion

**Review by rancher-max — COMMENTED** [review]
I didn't confirm versions on stuff yet but this looks good to get us started. 

I think we also need to add in v1.23.4 now with a minChannelServerVersion of v2.6.4-alpha1

**Comment by snasovich** [comment]
Piggyback on @rancher-max 's comment, we also want to change the default version to v1.23 as well.
Also, wondering why we're adding RCs to KDM?

**Comment by rancher-max** [comment]
@snasovich to answer the question about adding RCs to KDM, it was a request from Craig and Denise a while ago. This is a snippet of our `Cut Release` issue template in rke2. 

```
- [ ] Release Captain: Prepare PRs as needed to update [KDM](https://github.com/rancher/kontainer-driver-metadata/) in the appropriate dev branches using an RC.  For more information on the structure of the PR, see the [docs](https://github.com/rancher/rke2/blob/master/developer-docs/upgrading_kubernetes.md#update-rancher-kdm)
  - [ ] If server args, agent args, or charts are changed, link relevant rancher/rancher issue or create new rancher/rancher issue
  - [ ] If any new issues are created, escalated to Rancher PJM so they know and can plan for it 
- [ ] EM: Review and merge above PR
- [ ] QA: Post merge, run rancher with KDM pointed at the dev branch (where the PR in the previous step was merged) and test import, upgrade, and provisioning against those RCs. This work may be split between Rancher and RKE2 QAs.
```

File: channels-rke2.yaml [hunk 3]
```diff
@@ -325,6 +325,33 @@ releases:
...
+    maxChannelServerVersion: v2.6.99
+    serverArgs: &serverArgs-v1-21-10-rc2-rke2r2
+      <<: *serverArgs-v1-21-10-rke2r1
+    agentArgs: &agentArgs-v1-21-10-rc2-rke2r2
+      <<: *agentArgs-v1-21-10-rke2r1
+    charts: &charts-v1-21-10-rc2-rke2r2
+      <<: *charts-v1-21-10-rke2r1
+      rke2-canal:
+        repo: rancher-rke2-charts
+        version: v3.21.4-build2022020801
```
Comment (rancher-max) [inline channels-rke2.yaml]: CI actually is helpful in this case! I would've never seen this (one of the 0's should be a 2 apparently). 
```suggestion
        version: v3.21.4-build2022022801
```

**Review by rancher-max — DISMISSED** [review]

**Review by rancher-max — DISMISSED** [review]

**Review by rancher-max — APPROVED** [review]

**Review by dereknola — APPROVED** [review]

File: channels-rke2.yaml [hunk 4]
```diff
@@ -381,4 +408,52 @@ releases:
         version: v3.21.4-build2022020801
       rke2-calico:
         repo: rancher-rke2-charts
-        version: v3.21.401
\ No newline at end of file
+        version: v3.21.401
+  - version: v1.22.7+rke2r2
+    minChannelServerVersion: v2.6.3-alpha1
```
Comment (ShylajaDevadiga) [inline channels-rke2.yaml:413]: Is it supposed to be v2.6.4-alpha1?

Comment (ShylajaDevadiga) [reply channels-rke2.yaml:413]: Never mind its expected to be 2.6.3 on v1.22

**Review by ShylajaDevadiga — APPROVED** [review]

**Pull request merged**